src/main/java/Duke.java

```diff
@@ -1,10 +1,275 @@
+        if (!userInput.matches(RegexType.EVENT_REGEX.getRegexType())){
+            throw new EventException();
+        }
+
+        // get description and by from userInput
+        String description = userInput.replaceAll(RegexType.GET_DESCRIPTION_REGEX.getRegexType(), "");
+        String at = userInput.replaceAll(RegexType.GET_AT_REGEX.getRegexType(), "");
+
+        // Create new Event instance an add it to end taskList
+        taskList.add(new Event (description, at));
+
+        printAcknowledgement(RegexType.TASK_ADDED_LINE.getRegexType(), taskList.size()-1);
+        printNumOfTask();
+    }
+
+    public static void addNothing() throws NothingException{
+        throw new NothingException();
+    }
+
+    // Delete line
+    public static void deleteTask(String userInput)
+            throws DeleteFormatException, DeleteRangeException {
+
+        // Check if the command is done and is followed by a number
+        // and if the index is within the range of number of tasks
+        if (!userInput.matches(RegexType.DIGITS_REGEX.getRegexType())){
+            throw new DeleteFormatException();
+        }
+
+        int index = Integer.parseInt(userInput) - 1;
+        if (index >= taskList.size()){
+            throw new DeleteRangeException();
+        }
+
+        printAcknowledgement(PrintType.TASK_DELETED_LINE.getPrintType(), index);
+
+        taskList.remove(index);
+
+        printNumOfTask();
+    }
+
+    // Mark the task at the given index as done
+    public static void markAsDone(String userInput)
+            throws DoneFormatException, DoneAlreadyException, DoneRangeException{
+
+        // Check if the command is done and is followed by a number
+        // and if the index is within the range of number of tasks
+        if (!userInput.matches(RegexType.DIGITS_REGEX.getRegexType())){
+            throw new DoneFormatException();
+        }
+
+        int index = Integer.parseInt(userInput) - 1;
+        if (index >= taskList.size()){
+            throw new DoneRangeException();
+        }
+        // Check if task is already done
+        if (taskList.get(index).isDone()){
+            throw new DoneAlreadyException();
+        }
+
+        // Mark the index as done
+        taskList.get(index).markAsDone();
+
+        // Acknowledge task is done
+        printAcknowledgement(PrintType.TASK_DONE_LINE.getPrintType(), index);
+    }
+
+    // Reads the input of the line to determine the command and run it
+    public static void readInput(String userInput){
+
+        // Get command from the userInput
+        String[] splitLine = userInput.split(" ", 2);
+        String command = splitLine[0];
+
+        // Remove command from userInput
+        userInput = userInput.replaceAll( RegexType.START_LINE_REGEX.getRegexType() + command + RegexType.SPACE_REGEX.getRegexType(), "");
+
+        // Check the command type then execute the commands
+        if(userInput.equals(InputType.LIST.getInputType())) {
```

Comment:
Could this list of checks better be served with a switch statement to improve readability?